compiler/rustc_errors/src/lib.rs

```diff
@@ -1340,9 +1348,22 @@ pub fn add_elided_lifetime_in_path_suggestion(
     );
 }
 
-// Useful type to use with `Result<>` indicate that an error has already
-// been reported to the user, so no need to continue checking.
+/// Useful type to use with `Result<>` indicate that an error has already
+/// been reported to the user, so no need to continue checking.
 #[derive(Clone, Copy, Debug, Encodable, Decodable, Hash, PartialEq, Eq)]
-pub struct ErrorReported;
+pub struct ErrorReported(InnerErrorReported);
```

Comment:
FWIW, I think usually we do something like `pub struct ErrorReported(())`. At least, that's what `ty::Error` uses IIRC.